config-service-api/src/main/proto/org/hypertrace/config/service/v1/config_service.proto

```diff
@@ -7,27 +7,90 @@ package org.hypertrace.config.service.v1;
 import "google/protobuf/struct.proto";
 
 service ConfigService {
+  // Merges the specified config with the existing config and upserts the
+  // resulting config into the store. Also returns the config which is upserted.
   rpc UpsertConfig(UpsertConfigRequest) returns (UpsertConfigResponse) {}
+
+  // Gets the config for the specified request
   rpc GetConfig(GetConfigRequest) returns (GetConfigResponse) {}
+
+  // Gets all the configs(i.e. across all the contexts) for the specified request
+  rpc GetAllConfigs(GetAllConfigsRequest) returns (GetAllConfigsResponse) {}
+
+  // Deletes the config for the specified request
+  rpc DeleteConfig(DeleteConfigRequest) returns (DeleteConfigResponse) {}
 }
 
 message UpsertConfigRequest {
-  string resource_name = 1;  // required
-  string resource_namespace = 2;  // required - namespace with which the config resource is associated
-  google.protobuf.Value config = 3; // required - config value to be upserted
-  string context = 4; // optional - only required if config applies to a specific context
+  // required - name of the resource associated with the config
+  string resource_name = 1;
+
+  // required - namespace with which the config resource is associated
+  string resource_namespace = 2;
+
+  // required - config value to be upserted
+  google.protobuf.Value config = 3;
+
+  // optional - only required if config applies to a specific context.
+  // If empty, specified config is associated with a default context.
+  string context = 4;
 }
 
 message UpsertConfigResponse {
-  int64 config_version = 1; // the version allocated to the newly inserted configuration
+  // config value upserted into the store
+  google.protobuf.Value config = 1;
```

Comment:
Given that this isn't used yet, we can make the breaking change - but the fact that the build passes indicates we haven't set up breaking change detection as part of CI yet - can we please do that in a follow up (this PR is already doing too much)

Reply:
Oh, yes. I had commented out the step for detecting breaking change in the first PR thinking that I would uncomment it in a subsequent PR but forgot to do so. Will definitely do it in a follow-up PR.